remoteconfig/condition_evaluator.go

```diff
@@ -0,0 +1,131 @@
+	evaluatedConditions := make(map[string]bool)
+	for _, condition := range ce.conditions {
+		evaluatedConditions[condition.Name] = ce.evaluateCondition(condition.Condition, rootNestingLevel)
+	}
+	return evaluatedConditions
+}
+
+func (ce *conditionEvaluator) evaluateCondition(condition *oneOfCondition, nestingLevel int) bool {
+	if nestingLevel >= maxConditionRecursionDepth {
+		log.Println("Maximum recursion depth is exceeded.")
+		return false
+	}
+
+	if condition.Boolean != nil {
+		return *condition.Boolean
+	} else if condition.OrCondition != nil {
+		return ce.evaluateOrCondition(condition.OrCondition, nestingLevel+1)
+	} else if condition.AndCondition != nil {
+		return ce.evaluateAndCondition(condition.AndCondition, nestingLevel+1)
+	} else if condition.Percent != nil {
+		return ce.evaluatePercentCondition(condition.Percent)
+	}
+	log.Println("Unknown condition type encountered.")
+	return false
+}
+
+func (ce *conditionEvaluator) evaluateOrCondition(orCondition *orCondition, nestingLevel int) bool {
+	for _, condition := range orCondition.Conditions {
+		result := ce.evaluateCondition(&condition, nestingLevel+1)
+		// short-circuit evaluation, return true if any of the conditions return true
+		if result {
+			return true
+		}
+	}
+	return false
+}
+
+func (ce *conditionEvaluator) evaluateAndCondition(andCondition *andCondition, nestingLevel int) bool {
+	for _, condition := range andCondition.Conditions {
+		result := ce.evaluateCondition(&condition, nestingLevel+1)
+		// short-circuit evaluation, return false if any of the conditions return false
+		if !result {
+			return false
+		}
+	}
+	return true
+}
+
+func (ce *conditionEvaluator) evaluatePercentCondition(percentCondition *percentCondition) bool {
+	if rid, ok := ce.evaluationContext[randomizationID].(string); ok {
+		if percentCondition.PercentOperator == "" {
+			log.Println("Missing percent operator for percent condition.")
+			return false
+		}
+		instanceMicroPercentile := computeInstanceMicroPercentile(percentCondition.Seed, rid)
+		switch percentCondition.PercentOperator {
+		case lessThanOrEqual:
+			return instanceMicroPercentile <= percentCondition.MicroPercent
+		case greaterThan:
+			return instanceMicroPercentile > percentCondition.MicroPercent
+		case between:
+			return instanceMicroPercentile > percentCondition.MicroPercentRange.MicroPercentLowerBound && instanceMicroPercentile <= percentCondition.MicroPercentRange.MicroPercentUpperBound
+		default:
+			log.Printf("Unknown percent operator: %s\n", percentCondition.PercentOperator)
+			return false
+		}
+	}
+	log.Println("Missing or invalid randomizationID (requires a string value) for percent condition.")
+	return false
+}
+
+func computeInstanceMicroPercentile(seed string, randomizationID string) uint32 {
+	seedPrefix := ""
+	if len(seed) > 0 {
+		seedPrefix = fmt.Sprintf("%s.", seed)
+	}
+	stringToHash := fmt.Sprintf("%s%s", seedPrefix, randomizationID)
+
```

Comment:
How long the seed and randomizationID can be? Would it make sense to optimize this with a String builder?

```go
var sb strings.Builder
if len(seed) > 0 {
  sb.WriteString(seed) // More efficient than fmt.Sprintf for simple appends
  sb.WriteRune('.')    // Efficient for single characters
}
sb.WriteString(randomizationID)
return sb.String()
```
Just a thought, It might not make sense if the strings are not significantly long :)

Reply:
`randomizationID` is passed by the user; could be arbitrarily long. 
Will update the PR to use the builder pattern.

Reply:
Have addressed in the PR where minor refactors have been made - https://github.com/firebase/firebase-admin-go/commit/8649eaae0d8252017fe459f242ddec8c2dc74bac